src/remux/passthrough-remuxer.js

```diff
@@ -2,24 +2,93 @@
  * passthrough remuxer
 */
 import Event from '../events';
+import { parseInitSegment, parseUserData, parseSei } from '../utils/mp4-tools';
 
 class PassThroughRemuxer {
   constructor (observer) {
     this.observer = observer;
+    this.initData = {};
+    this.initPTS = 0;
+    this.initSegment = {};
+    this.initTracks = {};
   }
 
   destroy () {
   }
 
-  resetTimeStamp () {
+  resetTimeStamp (defaultInitPTS) {
+    this.initPTS = defaultInitPTS;
+    this.lastEndDTS = null;
   }
 
-  resetInitSegment () {
+  resetNextTimestamp () {
+    this.lastEndDTS = null;
+  }
+
+  resetInitSegment (initSegment, audioCodec, videoCodec) {
+    this.audioCodec = audioCodec;
+    this.videoCodec = videoCodec;
+    this.generateInitSegment(initSegment);
+    this.emitInitSegment = true;
+  }
+
+  generateInitSegment (initSegment) {
+    let { audioCodec, videoCodec } = this;
+    if (!initSegment || !initSegment.byteLength) {
+      this.initTracks = undefined;
+      this.initData = undefined;
+      return;
+    }
+    const initData = this.initData = parseInitSegment(initSegment);
+
+    // default audio codec if nothing specified
+    // TODO : extract that from initsegment
```

Comment:
No, it was from copying a method in the 1.0.x branch because I needed info from init segment and there was no init segment in the passthrough class in 0.12